primedev/client/debugoverlay.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
```

Comment:
I'm very much considering doing a quick PR with just that change of moving typedefs to header file ^^

Reply:
typedefs in header files just clutter up your intelisense. Using typedefs also means intellisense gives you the type only, not the signature when typing out a function call.

If you really want to just move all function pointers into headers and get rid of all typedefs that aren't used as arguments in a function.

Reply:
Wait, so why were they moved to header files in the first place then? Or were you just experimenting around at that point? ^^

Reply:
I mean TBH typedefs shouldn't be used in 99% of cases anyway

Reply:
> Wait, so why were they moved to header files in the first place then? Or were you just experimenting around at that point? ^^

I needed to move the function pointers to the header fie so i could use them elsewhere. This meant i had to also move the typedefs.

Reply:
I have now removed the typedefs and just used decltype where needed